### ISSUE TITLE
onError do nothing

### ISSUE THREAD

**Issue by zbyso23** [vankop/jsoneditor-react#17]
I try to use onError, but do nothing, and when I look to implementation - is logically then do nothing because is not implemented :)
is planned to implement this useful property? thanks

**Comment by vankop** [comment]
@zbyso23 PR wellcome.

**Comment by zbyso23** [comment]
> @zbyso23 PR wellcome.

okay, how to? I don't have experience with collaboration on GitHub project, trying create new branch - handle-errors but commit is impossible - don't have `correct access rights`

**Comment by emmanuelgautier** [comment]
You have to fork this project, then push the new branch you created into your newly created repo. After that, you should be able to create a new PR from the Github interface.

**Comment by zbyso23** [comment]
thanks - I try to implement subscribe changeAnnotation, but I noticed strange behaviour of Ace editor - when is fired in callback, return empty object - but, I say "okay, I get manually", but when i call directly code like this: 
```
const annotations = this.jsonEditor.aceEditor.getSession().getAnnotations();
```
in annotations return OLD error! then I fix for now by code like this:
```
setTimeout(() => {
   const annotations = this.jsonEditor.aceEditor.getSession().getAnnotations();
}, 150);
```
but is very dirt - time can be not enough on old computers :/
thrid issue what i registrer, even i call this code only once:
```
this.jsonEditor.aceEditor.getSession().on('changeAnnotation', this.handleError);
```
is every time called twice...
if anybody notice similar behaviour i very thanks for any help :)

**Comment by vankop** [comment]
@zbyso23 You need to create issue in upstream dependency

**Comment by vankop** [comment]
in `jsoneditor@9` there is an option `onError`, so current prop `onError` will be passed in editor options

**Comment by AnnaRainier** [comment]
this onError method fires on my component only if I trigger "filter, sort or transform contents" manually, I expected it to fire once an error occurs in JSON file. If I need to rely on JSON validity state, for, let's say, some button disabled state, how do I do that?

**Comment by zbyso23** [comment]
I solve this problem by code above (10 Jun 2020), but it's not part of current code - i use standalone file `Editor.tsx` in my project

**Comment by Isaacgl68** [comment]
you can use the onValidationError of the json editor
You just need to add it in the editor props (it will pass to the jeson editor as the editor is passing the ...rest props)

const onValidationError = (value)=>{ 
        if (value && value.length > 0) {
          //do 
        }
    }
<Editor
...
onValidationError={onValidationError} />
